### PR TITLE
da1469x: otp and serial script enhancements

### PR DESCRIPTION
**Pull request by nkaje**
This PR provides script enhancements to 

1. da146x_serial.py: Makes the reset script optional. Use manual reset method or script to enter serial load mode.
2. da1469x_serial.py: Provide a common function that flushes serial port when initially opened.
3. otp_tool.py: Provide a common function that flushes serial port when initially opened.

### CODE REVIEW

**Comment by nkaje** [comment]
@agross-korg @benmccrea Please review.

File: hw/bsp/dialog_da1469x-dk-pro/da1469x_serial.py [hunk 1]
```diff
@@ -35,6 +35,15 @@ def load(infile, uart, reset_script):
...
 
+    # drain serial port buffer
+    try:
+        while True:
+            data = ser.read(1)
+            if len(data) == 0:
+                break
+    except serial.SerialException:
+        raise SystemExit("Failed to open serial port")
+
```
Comment (agross-korg) [inline hw/bsp/dialog_da1469x-dk-pro/da1469x_serial.py:46]: nit.  you're mixing two sets of changes here.  One to drain the port and the other to make the reset optional.

Comment (nkaje) [reply hw/bsp/dialog_da1469x-dk-pro/da1469x_serial.py:46]: done.

**Comment by nkaje** [comment]
Test logs
Manual reset:
```
./repos/apache-mynewt-core/hw/bsp/dialog_da1469x-dk-pro/da1469x_serial.py load -u /dev/ttyUSB1 ./bin/targets/ram_executable/ram_executable.elf.bin 
Please reset board to enter ROM uart recovery
b'\x02'
Detected serial boot protocol
Loading application to RAM
Successfully loaded RAM, board will now boot executable
```
Communicate with the target:
```
$ ./repos/apache-mynewt-core/hw/bsp/dialog_da1469x-dk-pro/otp_tool.py test-alive-target -u /dev/ttyUSB0
Successfully communicated with target
```
and performed other one time programmable routines.

**Pull request merged**